modules/olm-arch-catalog-operator.adoc

```diff
@@ -0,0 +1,25 @@
+// Module included in the following assemblies:
+//
+// * operators/understanding_olm/olm-arch.adoc
+
+[id="olm-arch-catalog-operator_{context}"]
+= Catalog Operator
+
+The Catalog Operator is responsible for resolving and installing CSVs and the
+required resources they specify. It is also responsible for watching
+CatalogSources for updates to packages in channels and upgrading them
+(optionally automatically) to the latest available versions.
+
+A user who wishes to track a package in a channel creates a Subscription
+resource configuring the desired package, channel, and the CatalogSource from
+which to pull updates. When updates are found, an appropriate InstallPlan is
+written into the namespace on behalf of the user.
+
+.Catalog Operator workflow
+* Has a connection to each CatalogSource in the cluster.
+* Watches for unresolved InstallPlans created by a user:
+** Finds the CSV matching the name requested and adds it as a resolved resource.
+** For each managed or required CRD, adds it as a resolved resource.
+** For each required CRD, finds the CSV that manages it.
```

Comment:
Having a hard time reading how the first-level bullet introduces the second-level bullets. There are a lot of levels to connect from workflow -> watches for InstallPlans -> finds/adds. For example, when I get to the part where I read "adds it a resolved resource", I have to think about what "it" is referring to. Is "it" the InstallPlan, the CSV, or the name?

Not sure how best to rework this...ordered list, maybe? Or paragraphs? Even if you write out each item as a complete sentence, I think it will be easier to follow along.

Reply:
Rewrote as complete sentences in an ordered list.